tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -3875,3 +3869,220 @@ def test_start_trace_with_invalid_experiment_id(store: SqlAlchemyStore):
+        # Order by status
+        (["status"], ["tr-1", "tr-4", "tr-3", "tr-0", "tr-2"]),
+        # Order by name
+        (["name"], ["tr-1", "tr-2", "tr-3", "tr-4", "tr-0"]),
+        # Order by tag (null comes last)
+        (["tag.fruit"], ["tr-2", "tr-1", "tr-3", "tr-4", "tr-0"]),
+        # Order by multiple tags
+        (["tag.fruit", "tag.color"], ["tr-2", "tr-1", "tr-3", "tr-4", "tr-0"]),
+        # Order by non-existent tag (should be ordered by default order)
+        (["tag.nonexistent"], ["tr-4", "tr-3", "tr-2", "tr-1", "tr-0"]),
+        # Order by run Id
+        (["run_id"], ["tr-0", "tr-1", "tr-4", "tr-3", "tr-2"]),
+    ],
+)
+def test_search_traces_order_by(store_with_traces, order_by, expected_ids):
+    exp1 = store_with_traces.get_experiment_by_name("exp1").experiment_id
+    exp2 = store_with_traces.get_experiment_by_name("exp2").experiment_id
+    trace_infos, _ = store_with_traces.search_traces(
+        experiment_ids=[exp1, exp2],
+        filter_string=None,
+        max_results=5,
+        order_by=order_by,
+    )
+    actual_ids = [trace_info.request_id for trace_info in trace_infos]
+    assert actual_ids == expected_ids
+
+
+@pytest.mark.parametrize(
+    ("filter_string", "expected_ids"),
+    [
+        # Search by name
+        ("name = 'aaa'", ["tr-1"]),
+        ("name != 'aaa'", ["tr-4", "tr-3", "tr-2", "tr-0"]),
+        ("name LIKE 'b%'", ["tr-2"]),
+        ("name ILIKE 'd%'", ["tr-4", "tr-0"]),
+        # Search by status
+        ("status = 'OK'", ["tr-4", "tr-3", "tr-0"]),
+        ("status != 'OK'", ["tr-2", "tr-1"]),
+        # Search by timestamp
+        ("`timestamp` >= 1 AND execution_time < 10", ["tr-2", "tr-1"]),
+        # Search by tag
+        ("tags.fruit = 'apple'", ["tr-2", "tr-1"]),
+        ("tags.fruit = 'apple' and tags.color != 'red'", ["tr-2"]),
+        ("tags.color LIKE 're%'", ["tr-1"]),
+        # Search by request metadata
+        ("run_id = 'run0'", ["tr-0"]),
+    ],
+)
+def test_search_traces_with_filter(store_with_traces, filter_string, expected_ids):
+    exp1 = store_with_traces.get_experiment_by_name("exp1").experiment_id
+    exp2 = store_with_traces.get_experiment_by_name("exp2").experiment_id
+
+    trace_infos, _ = store_with_traces.search_traces(
+        experiment_ids=[exp1, exp2],
+        filter_string=filter_string,
+        max_results=5,
+        order_by=[],
+    )
+    actual_ids = [trace_info.request_id for trace_info in trace_infos]
+    assert actual_ids == expected_ids
+
+
+def test_search_traces_raise_if_max_results_arg_is_invalid(store):
+    with pytest.raises(MlflowException, match="Invalid value for request parameter"):
+        store.search_traces(experiment_ids=[], max_results=50001)
+
+    with pytest.raises(MlflowException, match="Invalid value for request parameter"):
+        store.search_traces(experiment_ids=[], max_results=-1)
+
+
+def test_search_traces_pagination(store_with_traces):
+    exps = [
```

Comment:
Very odd test error here. Perhaps call for the experiments outside of the collection and then create the list?

Reply:
@BenWilson2 Yea I thought it was some flakiness but it seems there are some intra-dependencies between test cases.